Source/DafnyDriver/Commands/VerifyCommand.cs

```diff
@@ -35,6 +43,13 @@ static VerifyCommand() {
   public static readonly Option<string> FilterPosition = new("--filter-position",
     @"Filter what gets verified based on a source location. The location is specified as a file path suffix, optionally followed by a colon and a line number. For example, `dafny verify dfyconfig.toml --filter-position=source1.dfy:5` will only verify things that range over line 5 in the file `source1.dfy`. In combination with `--isolate-assertions`, individual assertions can be verified by filtering on the line that contains them. When processing a single file, the filename can be skipped, for example: `dafny verify MyFile.dfy --filter-position=:23`");
 
+  public static readonly Option<string> ExtractTarget = new("--extract", @"
+Extract Dafny types, functions, and lemmas to Boogie.
```

Comment:
What does this do? It's still not clear to me how this is different from `--bprint`. Does this ignore the input program and only output something relates to Dafny internals? If so, why is it part of `dafny verify` ?

Reply:
Extraction is the output that the Dafny tool will incorporate into `DafnyPrelude.bpl`. These are not verification conditions but axioms and other definitions. In contrast, `--bprint` shows the verification conditions that are sent to Boogie.

To accomplish its task, extraction pays attention to `:extract_boogie_name` attributes and renames things accordingly. Extraction turns lemmas into universally quantified axioms (whereas VC generation turns them into Boogie procedures). Finally, extraction looks at the input Dafny expressions as they were Boogie expressions, more or less. So, the extraction output is free of well-formedness conditions and `$Is` predicates and other such things that are part of VC generation.